cluster-provision/k8s/1.17/provision.sh

```diff
@@ -2,19 +2,27 @@
 
 set -ex
 
+if [ ! -f "/tmp/extra-pre-pull-images" ]; then
+    echo "WARNING: extra-pre-pull-images list missing"
```

Comment:
Shouldn't we fail the process if we failed to pre-pull the images?

Reply:
@oshoval argued against this as to be able to iterate faster when provisioning clusters locally, which makes sense to me.

Reply:
i meant please if a flag is set SKIP_PULL or something, then it would ?

Reply:
Why do we need a warning if we have no images to pull? Let's either agree on a required empty file, or no warning since the sanity check afterwards would detect missing images. Am I missing something?

Reply:
the sanity indeed also need to know what it should check, 
i think that it should enforce the file and the check, unless there is a flag that tell them both, all good pass on

Reply:
Required empty file makes sense. Then let's fail fast if it's not present.